parity/cache.rs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015, 2016 Ethcore (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+/// Configuration for application cache sizes.
+/// All	values are represented in MB.
+#[derive(Debug, PartialEq)]
+pub struct CacheConfig {
+	/// Size of rocksdb cache set using option `set_block_cache_size_mb`
+	/// 50% is blockchain
+	/// 25% is tracing
+	/// 25% is state
+	pub rocksdb: u32,
+	/// Size of blockchain cache.
+	pub blockchain: u32,
+	/// Size of transaction queue cache.
+	pub queue: u32,
+}
+
+impl Default for CacheConfig {
+	fn default() -> Self {
+		CacheConfig::new_with_total_cache_size(200)
+	}
+}
+
+impl CacheConfig {
+	/// Creates new cache config with cumulative size equal `total`.
+	pub fn new_with_total_cache_size(total: u32) -> Self {
+		CacheConfig {
+			rocksdb: total / 2,
+			blockchain: total / 4,
```

Comment:
This should default to a few megabytes. Querying old blocks is an extremely rare case.